src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs

```diff
@@ -32,8 +32,26 @@ public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode nod
             var pn = on.Children.OfType<XamlAstXamlPropertyValueNode>()
                 .FirstOrDefault(p => p.Property.GetClrProperty().Name == "Selector");
 
+            // Missing selector, use the object's target type if available
             if (pn == null)
+            {
+                // We already went through this node
+                if (context.ParentNodes().FirstOrDefault() is AvaloniaXamlIlTargetTypeMetadataNode metadataNode
+                    && metadataNode.Value == on)
+                {
+                    return node;
+                }
+
+                if (FindStyleParentObject(context) is { } parentObjectNode)
```

Comment:
How would it work, if Style without selector is inside of ControlTheme?

Reply:
Of I suppose ControlTheme case would be already handled at this point above. Same with nested styles. Would make sense to add test case for these.

Other than that, I think lookup for StyledObject would make sense here. But can be too restrictive in some corner cases.

Reply:
Good catch.

Styles without a selector in `ControlTheme` weren't failing at compile time (even before this PR), since we're in a `Style` scope with a proper target type thanks to `AvaloniaXamlIlControlThemeTransformer`. They were failing at runtime though.

I've made sure the XAML compiler now handles this case properly, with proper unit tests. I've also restricted the parent to `StyledElement`.